kernel/src/drivers/allocator.c

```
@@ -0,0 +1,145 @@
+	return (void*)(search + 1);
+}
+
+void* realloc(void* ptr, size_t new_size) {
+	uint8_t* buffer = malloc(new_size);
+
+	MemoryHeader_t* old_header = __get_header(ptr);
+
+	// copy old data to new buffer
+	size_t memory_to_copy = new_size > old_header->size ? old_header->size : new_size;
+	for (size_t i = 0; i < memory_to_copy; i++) {
+		buffer[i] = ((uint8_t*)ptr)[i];
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)